pulsar-broker/src/main/java/org/apache/pulsar/broker/service/DrainingHashesTracker.java

```diff
@@ -252,24 +322,40 @@ public synchronized void reduceRefCount(Consumer consumer, int stickyHash, boole
                 log.debug("[{}] Draining hash {} removing consumer id:{} name:{}", dispatcherName, stickyHash,
                         consumer.consumerId(), consumer.consumerName());
             }
-            DrainingHashEntry removed = drainingHashes.remove(stickyHash);
+            DrainingHashEntry removed;
+            lock.writeLock().lock();
+            try {
+                removed = drainingHashes.remove(stickyHash);
+            } finally {
+                lock.writeLock().unlock();
```

Comment:
is this necessary to lock and release this quick?

Reply:
Fair point. I'll also handle the "unblock" accesses during the same acquired lock.